### PR TITLE
Feature/search in listings

### PR DESCRIPTION
**Pull request by krysal**
## Description
This PR enables the search input of each listing. Initially, search for the user input in these fields:
- For **cases**: name, courts, related_cases, background, summary.
- For **scholarships**: title, authors, summary

## Deployment preview
See https://caselaw-mirr-feature-se-gqlssn.herokuapp.com/
Check for `/cases` & `/scholarship` paths.

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: legal_db/views.py [hunk 1]
```diff
@@ -20,15 +21,31 @@ def get_context_data(self, **kwargs):
...
+        keywords = self.request.GET.get("keywords")
+        if keywords:
+            attributes = [
+                "name",
+                "courts",
+                "related_cases",
+                "background",
+                "summary",
+            ]
+            f = build_filters(attributes, keywords)
```
Comment (kgodey) [inline legal_db/views.py:40]: Please avoid using single letter variable names, it's very unclear.

**Comment by krysal** [comment]
@kgodey  I looked to `django-filter` package and I found it very useful to make detailed filters for models, but for this particular case I think it doesn't quite fit, the way it works creates a form with an individual input field for each filter, but how I pictured this feature was to use a single input field to search on multiple attributes and return the row if a match if found for any of those (OR condition).
This behavior appears to be a work in progress currently: carltongibson/django-filter#1134, carltongibson/django-filter#1167

Although I think it may be useful for the filtering by tags and push filter code on a file apart from `views.py`, I will review that case and make the correction of the variable name.

**Comment by kgodey** [comment]
@krysal that makes sense, I'll approve once you update the variable name.

**Comment by krysal** [comment]
Done :)

**Pull request merged**